executor/src/witgen/generator.rs

```diff
@@ -66,9 +103,44 @@ impl<'a, T: FieldElement> Generator<'a, T> {
         }
     }
 
-    pub fn run<Q: QueryCallback<T>>(&mut self, mutable_state: &mut MutableState<'a, '_, T, Q>) {
+    fn name(&self) -> &str {
+        let first_witness = self.witnesses.iter().next().unwrap();
+        let first_witness_name = self.fixed_data.column_name(first_witness);
+        let namespace = first_witness_name
+            .rfind('.')
+            .map(|idx| &first_witness_name[..idx]);
+        if let Some(namespace) = namespace {
+            namespace
+        } else {
+            // For machines compiled using Powdr ASM we'll always have a namespace, but as a last
+            // resort we'll use the first witness name.
+            first_witness_name
+        }
+    }
+
+    pub fn run<'b, Q: QueryCallback<T>>(&mut self, mutable_state: &mut MutableState<'a, 'b, T, Q>) {
```

Comment:
Can you briefly document the difference between `run` and `process_plookup`?

Reply:
Added a small comment!